pkg/auth/providers/ldap/ldap_provider.go

```diff
@@ -111,18 +112,32 @@ func (p *ldapProvider) TransformToAuthProvider(authConfig map[string]interface{}
 	return ldap, nil
 }
 
-func (p *ldapProvider) AuthenticateUser(ctx context.Context, input interface{}) (v3.Principal, []v3.Principal, string, error) {
+func toBasicLogin(input interface{}) (*v32.BasicLogin, error) {
 	login, ok := input.(*v32.BasicLogin)
 	if !ok {
-		return v3.Principal{}, nil, "", errors.New("unexpected input type")
+		return nil, errors.New("unexpected input type")
+	}
+	return login, nil
+}
+
+func (p *ldapProvider) AuthenticateUser(ctx context.Context, input interface{}) (v3.Principal, []v3.Principal, string, error) {
```

Comment:
While we're in here, since `AuthenticateUser` is exported it needs a docstring. (This is a new rule being applied to old code, but we should follow it for any function we change during a refactor like this.)